### ISSUE TITLE
Michelin - O365 Custom Learning failing

### ISSUE THREAD

**Issue by joshsullins** [pnp/custom-learning-office-365#23]
our tenant michelindev.onmicrosoft.com meets the prerequisites for O365 Custom Learning deployment, but it fails for us.

**Comment by karuanag** [comment]
Hello @joshsullins - Can you provide more details here about the error messages or what portion of the solution fails to complete?  That will help us understand how to route the issue.  @juliemturner if you already have this from speaking with the customer can you document here please? Thank you!

**Comment by joshsullins** [comment]
Already working with Matt Wolodarsky on this. See last response in the thread pasted below:



My answers in blue.


Josh SULLINS

Groupe Michelin | Information Technology
Global Collaboration Services Manager (End User eXperience Team)

[cid:image001.jpg@01D4A443.D1E38520]8 Rue de la Groliere, 63100 Clermont-Ferrand, France

[cid:image002.jpg@01D4A443.D1E38520]+33 07 68 83 77 82

[Monthly calendar]<https://outlook.office365.com/owa/calendar/df8ebcbd0bce4bddb0e635e101e532b4@michelin.com/12ae3f1ed17943faa7543a75209b1b8714839000025805348817/calendar.html>Calendar free/busy lookup<https://outlook.office365.com/owa/calendar/df8ebcbd0bce4bddb0e635e101e532b4@michelin.com/12ae3f1ed17943faa7543a75209b1b8714839000025805348817/calendar.html>


This e-mail may contain confidential information and is intended solely for the use of the addressee. Any disclosure of this information is strictly prohibited and may be unlawful. If you have received this e-mail by mistake, please notify us immediately and delete this e-mail.

From: Matt Wolodarsky <Matt.Wolodarsky@microsoft.com<mailto:Matt.Wolodarsky@microsoft.com>>
Sent: Monday, April 8, 2019 4:59 PM
To: Josh Sullins <josh.sullins@michelin.com<mailto:josh.sullins@michelin.com>>; Lanie Finlayson <lanie.finlayson@microsoft.com<mailto:lanie.finlayson@microsoft.com>>; Mike Hollinshead <mikeholl@microsoft.com<mailto:mikeholl@microsoft.com>>; Sébastien Portal <sportal@microsoft.com<mailto:sportal@microsoft.com>>; Julie Turner <julie.turner@sympraxisconsulting.com<mailto:julie.turner@sympraxisconsulting.com>>
Cc: Patrick Patterson <patrickp@microsoft.com<mailto:patrickp@microsoft.com>>; Eric Vigesaa <ericvig@microsoft.com<mailto:ericvig@microsoft.com>>; Sesha Mani <Sesha.Mani@microsoft.com<mailto:Sesha.Mani@microsoft.com>>; Peter Krebs <pkrebs@exchange.microsoft.com<mailto:pkrebs@exchange.microsoft.com>>
Subject: [EXT] RE: Michelin customer EBC and their feedback on 'Custom Learning for O365' package

Hi @Josh Sullins<mailto:josh.sullins@michelin.com> – can you please advise/run through the following troubleshooting steps?


  1.  What path did they take to install this… PnP provisioning or manual PnP provisioning
     *   If PnP provisioning, did they have to configure the app catalog or did it already exist, and they had to create it are they sure they waited the 30 minutes at least before trying to provision. The appcatalog already existed since months.
  2.  Is the webpart in the gallery? (if yes continue to step 3) Yes. See screenshot below.
  3.  Please navigate to the admin page (CustomLearningAdmin.aspx) in the pages library of the custom learning site collection you identified either when you provisioned or when you did the manual setup. Done, same result, see screenshot below #2.
     *   If using IE, please try Chrome browser. I only ever use Chrome.
     *   If still doesn’t work, validate that the user that’s doing the browsing is either a) site collection admin or b) owner of the site User is global admin and site collection admin and owner of the site.
If still doesn’t work, using Chrome, either a) take screen capture of console log showing any red errors or b) export the log by right mouse clicking and choosing “save as” and send those to us  How do I get to the console log?



[cid:image008.png@01D4EF8E.C7285250]

[cid:image009.png@01D4EF8E.C7285250]


Josh SULLINS

Groupe Michelin | Information Technology
Global Collaboration Services Manager (End User eXperience Team)

[cid:image001.jpg@01D4A443.D1E38520]8 Rue de la Groliere, 63100 Clermont-Ferrand, France

[cid:image002.jpg@01D4A443.D1E38520]+33 07 68 83 77 82

[Monthly calendar]<https://outlook.office365.com/owa/calendar/df8ebcbd0bce4bddb0e635e101e532b4@michelin.com/12ae3f1ed17943faa7543a75209b1b8714839000025805348817/calendar.html>Calendar free/busy lookup<https://outlook.office365.com/owa/calendar/df8ebcbd0bce4bddb0e635e101e532b4@michelin.com/12ae3f1ed17943faa7543a75209b1b8714839000025805348817/calendar.html>


This e-mail may contain confidential information and is intended solely for the use of the addressee. Any disclosure of this information is strictly prohibited and may be unlawful. If you have received this e-mail by mistake, please notify us immediately and delete this e-mail.

**Comment by peterjkrebs** [comment]
Took some sensitive info out of the thread.

**Comment by WonderLaura** [comment]
Hey @karuanag and @juliemturner !  Mine is failing, but I don't know where to look for any errors.  The comm site is getting created, but none of the custom learning is in there, just empty pages.  I even tried manually installing the app in the app catalog and in the comm site, and ran the powershell script provided in the manual instructions, although I hadn't initially started it as a manual installation.

**Comment by juliemturner** [comment]
@WonderLaura -- To summarize what I think is happening with a subset of people is one of two scenarios when provisioning. 
1. They didn't have or didn't wait long enough after creating their tenant app catalog and therefore the provisioning process silently fails to not put the web parts on the pages. 
2. Under some circumstances even with the app catalog previously provisioned the deployment process takes longer than than the scripting process and the web parts aren't available "yet" to be added to the pages.  

The flaw in the process is that even if you run provisioning again the pages already exists and so it's not recreated. To correct for this you need to first delete the pages and then re-provision or run the CustomLearningConfiguration.ps1 script in the webpart folder. In all cases so far where the web part shows up in the gallery but the admin page shows nothing on it, this has fixed the issue.

**TL;DR;**
This solution only fixes an instances where the CustomLearningAdmin.aspx page is COMPLETELY blank.  If you have a red banner telling you something is not configured properly or you do not have permissions this will not solve that issue.

1. In the custom learning site collection, navigate to the Site Pages library and delete _**CustomLearningAdmin.aspx**_ and **_CustomLearningViewer.aspx_**.
2. Run the [CustomLearningConfiguration.ps1](https://github.com/pnp/custom-learning-office-365/tree/master/webpart) PowerShell script which will recreate those pages with the web parts.

Also, we will be modifying the _CustomLearningConfiguration.ps1_ script shortly to delete the pages for you.

**Comment by WonderLaura** [comment]
Thanks @juliemturner , that worked!

**Comment by cueball59** [comment]
@juliemturner has the script been updated to automatically delete these? I have a call shortly with a customer that may be experiencing this.

**Comment by ToddKlindt** [comment]
> @juliemturner has the script been updated to automatically delete these? I have a call shortly with a customer that may be experiencing this.

Yes, it does. Technically it moves them to the Recycle Bin, but it gets them out of the way.
